src/net.cpp

```diff
@@ -86,7 +86,8 @@ std::map<CNetAddr, LocalServiceInfo> mapLocalHost GUARDED_BY(cs_mapLocalHost);
 static bool vfLimited[NET_MAX] GUARDED_BY(cs_mapLocalHost) = {};
 std::string strSubVersion;
 
-limitedmap<uint256, int64_t> mapAlreadyAskedFor(MAX_INV_SZ);
+limitedmap<uint256, int64_t> tx_to_ask(MAX_INV_SZ);
+std::map<uint256, std::multimap<int, std::pair<CNode*, CInv>>> tx_to_ask_from;
```

Comment:
Can this `CNode*` pointer dangle?

Reply:
It seems CNode constructor is deleted, and using a pointer is widely used across the codebase. What exactly would you suggest?

Reply:
That's indeed a good idea.